source/extensions/outlier_detection_monitors/consecutive_errors/consecutive_errors_monitor.cc

```diff
@@ -0,0 +1,50 @@
+#include "source/extensions/outlier_detection_monitors/consecutive_errors/consecutive_errors_monitor.h"
+
+#include "envoy/extensions/outlier_detection_monitors/consecutive_errors/v3/consecutive_errors.pb.h"
+#include "envoy/extensions/outlier_detection_monitors/consecutive_errors/v3/consecutive_errors.pb.validate.h"
+#include "envoy/registry/registry.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Outlier {
+
+bool ConsecutiveErrorsMonitor::onError() {
+  if (counter_ < max_) {
```

Comment:
This if block does not avoid race conditions, e.g. you have counter_ == 29 and max_ == 30:
* T1 reads the counter_ value in line 12 
* T2 increments counter_ value to 30
* T1 executes line 13 and increases counter_ to 31

Reply:
You could utilize atomic CAS primitive: https://en.cppreference.com/w/cpp/atomic/atomic_compare_exchange

Reply:
Yeah! That is good point. Will change the code.

Reply:
Done. Converted to using CAS primitives. Thanks for catching it!